roxie/ccd/ccdstate.cpp

```diff
@@ -966,69 +971,87 @@ class CRoxieQuerySetManager : public CInterface, implements IRoxieQuerySetManage
+                    const char *packageName = query->queryProp("@package");
+                    if (packageName && *packageName)
+                    {
+                        package = packages.queryPackage(packageName); // if a package is specified, require exact match
+                        if (!package)
+                            throw MakeStringException(ROXIE_QUERY_MODIFICATION, "Package %s specified by query %s not found", packageName, id);
+                    }
+                    else
+                    {
+                        package = packages.queryPackage(id);  // Look for an exact match, then a fuzzy match, using query name as the package id
+                        if(!package) package = packages.matchPackage(id);
+                        if (!package) package = &queryRootRoxiePackage();
+                    }
+                    assertex(package && QUERYINTERFACE(package, const IRoxiePackage));
+                    addQuery(id, loadQueryFromDll(id, queryDll.getClear(), *QUERYINTERFACE(package, const IRoxiePackage), query, forceRetry), queryHashes[i]);
                 }
-                else
+                catch (IException *E)
                 {
-                    package = packages.queryPackage(id);  // Look for an exact match, then a fuzzy match, using query name as the package id
-                    if(!package) package = packages.matchPackage(id);
-                    if (!package) package = &queryRootRoxiePackage();
+                    // we don't want a single bad query in the set to stop us loading all the others
+                    StringBuffer msg;
+                    msg.appendf("Failed to load query %s from %s", id ? id : "(null)", dllName ? dllName : "(null)");
+                    EXCLOG(E, msg.str());
+                    if (id)
+                    {
+                        StringBuffer emsg;
+                        E->errorMessage(emsg);
+                        Owned<IQueryFactory> dummyQuery = loadQueryFromDll(id, NULL, queryRootRoxiePackage(), NULL, false);
+                        dummyQuery->suspend(emsg.str());
+                        addQuery(id, dummyQuery.getClear(), queryHashes[i]);
+                    }
+                    E->Release();
                 }
-                assertex(package && QUERYINTERFACE(package, const IRoxiePackage));
-                addQuery(id, loadQueryFromDll(id, queryDll.getClear(), *QUERYINTERFACE(package, const IRoxiePackage), &query, forceRetry), hash);
-            }
-            catch (IException *E)
-            {
-                // we don't want a single bad query in the set to stop us loading all the others
-                StringBuffer msg;
-                msg.appendf("Failed to load query %s from %s", id ? id : "(null)", dllName ? dllName : "(null)");
-                EXCLOG(E, msg.str());
-                if (id)
+            });
+            for (auto h : queryHashes)
+                hash = rtlHash64Data(sizeof(h), &h, hash);
```

Comment:
I think this code is correct (what the function should have been doing), but it will not generate the same hash as before.  Is that an issue?

Reply:
no